File: src/main/java/frc/robot/subsystems/Shooter.java
package frc.robot.subsystems;

import edu.wpi.first.wpilibj.VictorSP;
import edu.wpi.first.wpilibj.command.Subsystem;

public class Shooter extends Subsystem {

    private static Shooter getShooter = new Shooter();

    public static Shooter getInstance() {
        return getShooter;
    }

    private VictorSP firstShooter, secondShooter;

    private Shooter() {
        firstShooter = new VictorSP(2);
        secondShooter = new VictorSP(3);
    }

    private static final double intakePower = 0.2;
    private static final double shootPower = 0.55;

    private void shootAtSpeed(double speed) {
        firstShooter.set(-speed);
        secondShooter.set(-speed);
    }

    public void intakeBall() {
        shootAtSpeed(0.15);
    }

    public void shootBall() {
        shootAtSpeed(shootPower);
    }

    public void reverseIntake() {
        shootAtSpeed(-intakePower);
    }

    public void notShooting() {
        firstShooter.set(0.0);
        secondShooter.set(0.0);
    }

    @Override
    public void initDefaultCommand() {

    }
}

File: src/main/java/frc/robot/subsystems/Shooter.kt
package frc.robot.subsystems

import edu.wpi.first.wpilibj.VictorSP
import frc.robot.Constants

private const val intakePower = 0.2
private const val shootPower = 0.55

object Shooter {

    private val leftShooter = VictorSP(Constants.leftShooter)
    private val rightShooter = VictorSP(Constants.rightShooter)

    private fun shootAtSpeed(speed: Double) {
        leftShooter.set(-speed)
        rightShooter.set(-speed)
    }


    fun intakeBall() {
        shootAtSpeed(0.15)
    }

    fun shootBall() {
        shootAtSpeed(shootPower)
    }

    fun reverseIntake() {
        shootAtSpeed(intakePower)
    }

    fun notShooting() {
        leftShooter.set(0.0)
        rightShooter.set(0.0)
    }

}